File: AllInOneKotlin/app/src/main/java/com/example/allinoneviewkotlin/MainActivity.kt
package com.example.allinonekotlin

import android.Manifest
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.core.content.ContextCompat
import com.trimblemaps.account.LicensedFeature
import com.trimblemaps.account.TrimbleMapsAccountManager
import com.trimblemaps.account.models.TrimbleMapsAccount
import com.trimblemaps.android.core.permissions.PermissionsListener
import com.trimblemaps.android.core.permissions.PermissionsManager
import com.trimblemaps.mapsdk.TrimbleMaps

class MainActivity : AppCompatActivity() {
    private val permissionsManager = PermissionsManager(null)
    private val permissionsListener = object: PermissionsListener {
        override fun onExplanationNeeded(permissionsToExplain: MutableList<String>?) {
        }

        override fun onPermissionResult(granted: Boolean) {
            // If location permissions were granted and a higher level
            // android version. Request background location permissions next.
            if(granted && PermissionsManager.isAndroidElevenPlus() &&
                !PermissionsManager.areLocationPermissionsGranted(this@MainActivity)) {
                permissionsManager.requestBackgroundLocationPermissions(this@MainActivity)
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val account = TrimbleMapsAccount.builder()
            .apiKey("Please reach out to Trimble MAPS for a license")
            .addLicensedFeature(LicensedFeature.NAVIGATION_SDK)
            .addLicensedFeature(LicensedFeature.MAPS_SDK)
            .build()

        TrimbleMapsAccountManager.initialize(account)
        TrimbleMapsAccountManager.awaitInitialization()
        TrimbleMaps.getInstance(applicationContext)

        if ((PermissionsManager.isAndroidElevenPlus() && ContextCompat.checkSelfPermission(this,
                Manifest.permission.ACCESS_BACKGROUND_LOCATION
            ) == 0)
            || ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == 0) {
            startActivity(Intent(this, AllInOneViewActivity::class.java))
        }
        else {
            permissionsManager.listener = permissionsListener
            permissionsManager.requestLocationPermissions(this)
        }
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        permissionsManager.onRequestPermissionsResult(requestCode, permissions, grantResults)

        if(PermissionsManager.isAndroidElevenPlus()) {
            if(permissions.isNotEmpty() && permissions.contains(Manifest.permission.ACCESS_BACKGROUND_LOCATION)) {
                startActivity(Intent(this, AllInOneViewActivity::class.java))
            }
        } else if (permissions.isNotEmpty() && permissions.contains(Manifest.permission.ACCESS_FINE_LOCATION)) {
            startActivity(Intent(this, AllInOneViewActivity::class.java))
        }
    }
}

File: maui/TrimbleMaps/MapSDK.Sample/MapSDK.Sample.Android/MainActivity.cs
using System.Threading.Tasks;
using Android.App;
using Android.Content.PM;
using Android.Runtime;
using Android.OS;
using Microsoft.Maui;
using Microsoft.Maui.ApplicationModel;

namespace TrimbleMaps.MapSDK.Sample.Droid
{
    using Com.Trimblemaps.Account.Models;
    using Com.Trimblemaps.Account;
    using Com.Trimblemaps.Mapsdk;
    using System.Collections.Generic;
    using Com.Trimblemaps.Android.Core.Permissions;
    using System.Linq;
    using CommunityToolkit.Maui.Core;
    using CommunityToolkit.Maui.Alerts;

    [Activity(Label = "TrimbleMaps.MapSDK.Sample", Icon = "@mipmap/icon", Theme = "@style/Maui.SplashTheme", MainLauncher = true, ConfigurationChanges = ConfigChanges.ScreenSize | ConfigChanges.Orientation | ConfigChanges.UiMode | ConfigChanges.ScreenLayout | ConfigChanges.SmallestScreenSize)]
    public class MainActivity : MauiAppCompatActivity
    {
        OnAccountLoaded _onAccountLoaded;
        PermissionsManager _permissionsManager = null;
        PermissionsListener _permissionsListener = null;

        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            Platform.Init(this, savedInstanceState);

            _permissionsListener = new PermissionsListener(this);
            _permissionsManager = new PermissionsManager(_permissionsListener);
            _permissionsManager.RequestLocationPermissions(this);

            _onAccountLoaded = new OnAccountLoaded(this);
        }

        private void InitializeTrimbleMaps()
        {
            var apiKey = "<API-KEY HERE>";

            var accountInitOptions = AccountInitializationOptions.InvokeBuilder()
                .Callback(_onAccountLoaded)
                //.Proxy(proxy)
                .Build();

            var account = TrimbleMapsAccount.InvokeBuilder()
                .ApiKey(apiKey)
                .AddLicensedFeature(LicensedFeature.MapsSdk)
                .Build();

            TrimbleMapsAccountManager.Initialize(account, accountInitOptions);
        }

        // On AccountInitializedCallback
        class OnAccountLoaded : Java.Lang.Object, TrimbleMapsAccountManager.IAccountInitializationListener
        {
            MainActivity _parent;
            public OnAccountLoaded(MainActivity parent)
            {
                _parent = parent;
            }

            public void OnAccountStatusChanged(AccountStatus status)
            {
                if (status == AccountStatus.Loaded)
                {
                    MainThread.BeginInvokeOnMainThread(() =>
                    {
                        Microsoft.Maui.Controls.Application.Current.MainPage.Navigation.PushAsync(new MainPage());
                        TrimbleMaps.GetInstance(_parent.Application.ApplicationContext);
                    });
                }
            }

            public void OnAccountStatusFailed(AccountStatus status, string message)
            {

                Toast.Make("Account status failed", ToastDuration.Short, 14).Show();

            }

            public void OnLicensingError(IList<LicensedFeature> p0)
            {

                Toast.Make("License status failed", ToastDuration.Short, 14).Show();

            }
        }

        class PermissionsListener : Java.Lang.Object, IPermissionsListener
        {
            MainActivity _parent;
            public PermissionsListener(MainActivity parent)
            {
                _parent = parent;
            }
            public void OnExplanationNeeded(IList<string> permissionsToExplain)
            {

            }

            public void OnPermissionResult(bool granted)
            {
                if (granted)
                {
                    Task.Run(() => _parent.InitializeTrimbleMaps());
                }
            }
        }

        public override void OnRequestPermissionsResult(int requestCode, string[] permissions, [GeneratedEnum] Android.Content.PM.Permission[] grantResults)
        {
            Platform.OnRequestPermissionsResult(requestCode, permissions, grantResults);

            if ((int)Build.VERSION.SdkInt >= 23)
            {
                base.OnRequestPermissionsResult(requestCode, permissions, grantResults);
            }

            _permissionsManager.OnRequestPermissionsResult(requestCode, permissions, grantResults.Cast<int>().ToArray());
        }
    }
}